src/StackedTable/extractHeaderLabels.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+import { StackedTableHead } from '../StackedTableHead.js'
+
+const isChildTableHead = child => child.type === StackedTableHead
```

Comment:
Will this work if I use a wrapped `StackedTableHead` component, or a custom component backed by a `thead`?

There's nothing special about the `StackedTableHead` so it should still work.

Reply:
I don't think this will be a problem, as these components will not be rendered, just used to extract the texts. The reason why these are components and not a prop is so you can configure colspans the "natural" way.

The `StackedTable` now accepts a `headerLabels` prop, so you could also just provide an array of strings instead, then this won't be used anymore, but it will only work for simpler cases. If we ever need to introduce rowspan in the header cells, it'd result in a very complex value we'd have to pass to the `StackedTable`.

If the app needs the header to be dynamic, you can still create the ui-core components through functions instead of components. It's not an ideal way, but other than that, I don't know how we should know which component is the actual header component.

Reply:
Ok, cool.